src/nimblepkg/nimscriptapi.nim

```diff
@@ -22,46 +25,157 @@ var
   foreignDeps*: seq[string] = @[] ## The foreign dependencies. Only
                                   ## exported for 'distros.nim'.
 
+  beforeHooks: seq[string] = @[]
+  afterHooks: seq[string] = @[]
+  commandLineParams: seq[string] = @[]
+  flags: TableRef[string, seq[string]]
+
+  command = "e"
+  project = ""
+  success = false
+  retVal = true
+
 proc requires*(deps: varargs[string]) =
   ## Call this to set the list of requirements of your Nimble
   ## package.
   for d in deps: requiresData.add(d)
 
+proc getParams() =
+  for i in 5 .. paramCount():
+    if paramStr(i).len != 0:
```

Comment:
``paramStr(i).len != 0`` should always be true.

Reply:
This was what caused the issue I mentioned. Empty string was coming in as a param. In fixed it in the caller so can be removed.

Reply:
Done